docs/pages/database-access/rbac/configuring-auto-user-provisioning.mdx

```diff
@@ -0,0 +1,166 @@
+```
+</TabItem>
+<TabItem label="Self-hosted PostgreSQL">
+Self-hosted PostgreSQL admin user must have X.509 authentication configured as
+described in the [self-hosted guide](../guides/postgres-self-hosted.mdx#step-35-configure-your-postgresql-server).
+```sql
+CREATE USER "teleport-admin" login createrole;
+```
+</TabItem>
+</Tabs>
+
+Users created by Teleport will be placed in the `teleport-auto-users` group in
+the database, which will be created automatically if it doesn't exist.
+
+Teleport will not delete the automatically created user at the end of the session.
+Instead, the user will be stripped of all roles, updated with `nologin` trait
+and reactivated during the next connection.
+
+Next, enable the database admin on the Teleport database configuration:
+
+<Tabs>
+<TabItem label="Static config">
+```yaml
+db_service:
+  enabled: "yes"
+  databases:
+  - name: "example"
+    protocol: "postgres"
+    uri: "localhost:5432"
+    admin_user:
+      name: "teleport-admin"
+```
+</TabItem>
+<TabItem label="Dynamic resource">
+```yaml
+kind: db
+version: v3
+metadata:
+  name: example
+spec:
+  protocol: "postgres"
+  uri: "localhost:5432"
+  admin_user:
+    name: "teleport-admin"
+```
+</TabItem>
+</Tabs>
+
+<Admonition type="tip" title="Auto-discovered databases">
+For auto-discovered cloud databases, the name of the admin user is taken from
+the `teleport.dev/db-admin` label.
+</Admonition>
+
+## Step 2/3. Configure Teleport role
+
+To specify the database roles a user should be assigned within the database,
+use `db_roles` role option:
+
+```yaml
+kind: role
+version: v6
+metadata:
+  name: auto-db-users
+spec:
+  options:
+    # create_db_user enables automatic user provisioning for matching databases
+    create_db_user: true
+  allow:
+    db_labels:
+      "*": "*"
+    db_names:
+    - "*"
+    # db_roles is a list of roles the database user will be assigned
+    db_roles:
+    - reader
+    - "{{internal.db_roles}}"
+    - "{{external.db_roles}}"
+```
+
+With automatic user provisioning, users always connect to the database with
+their Teleport username so the `db_users` role field is ignored for roles
+that have database user provisioning enabled.
+
+The `db_roles` role option supports all the same [interpolation rules](../../access-controls/guides/role-templates.mdx#interpolation-rules)
+as other role fields.
+
+User created within the database will:
+
+- Have the same name as Teleport username.
+- Be a part of the `teleport-auto-users` role.
+- Assigned all roles from the Teleport user's roleset that match the database.
+  The role names must be valid and exist in the database. See PostgreSQL
+  [CREATE ROLE](https://www.postgresql.org/docs/current/sql-createrole.html)
+  documentation for details.
+
+Note that in case of the name conflict where the user with the same name already
+exists in the database and is not managed by Teleport (i.e. not a part of the
+`teleport-auto-users` group), the connection will be aborted.
```

Comment:
```suggestion
Note that in case of a name conflict where a user with the same name already
exists in the database and is not managed by Teleport (i.e. not a part of the
`teleport-auto-users` group), the connection will be aborted.
```